### ISSUE TITLE
Memory leak in simulation

### ISSUE THREAD

**Issue by myliu-hub** [cepc/CEPCSW#205]
After testing, it was found that there was a memory leak problem when running the simulation.
![Uploading 20c28d1c5a7f5d72a02aa49eb8c8323.png…]()

**Comment by myliu-hub** [comment]
![20c28d1c5a7f5d72a02aa49eb8c8323](https://user-images.githubusercontent.com/59731907/158090306-b0522e50-2e34-424b-8e83-e5b920d92b26.png)

**Comment by mirguest** [comment]
Hi @myliu-hub ,
Thanks for reporting this issue. 

Could you explain a bit about your plot? What does x-axis mean? Could you disable some sub-detectors to debug this problem?

Because I don't have your source code, I try to measure the memory usage using the ECAL Matrix example. I can confirm that there is memory leakage in the simulation. In my example, the memory wil be increased about 20MB every 1k events. This problem could be caused by the creation of hit objects in either Geant4 Sensitive Detecotor or the conversion from Geant4 hit to EDM4hep hit. 

You can use following functions to monitor the memory easily:

```bash
mom ()
{
    local pid=$1;
    shift;
    local sleeptime=${1:-5};
    shift;
    while ps --pid $pid -o vsz,rss h; do
        sleep $sleeptime;
    done
}
```

Just pass the PID of gaudirun.py to `mom`, for example:
```bash
pid=$(ps aux | grep gaudirun | grep -v grep | awk '{print $2}') ; mom $pid >& mymem.$pid.txt
```

Tao

**Comment by mirguest** [comment]
This issue is fixed in PR #230.
